### DISCUSSION TITLE
Custom LaTeX.dic

### DISCUSSION THREAD

**Discussion by anton-vrba** [WinEdt-Team/WinEdt-Support#48] [1.1 General Help and Discussion]
(1) Can I set up a custom autocomplete dictionary to provide commands from loaded packages packages, but not autoloaded by WinEdt but manually loaded by the user

For examples: `shortintertext` from mathtools
also useful would be commands with options already filled, eg `\enlargethispage{1\baselineskip}`  which would be the obvious option in most cases. 

I can add these changes to the default C:\Program Files\WinEdt Team\WinEdt 11\Dict\LaTeX\LaTeX.dic  and that works, but is an ugly solution.

I tried various combinations creating adding a file to ...\AppData\Roaming\WinEdt Team\WinEdt 11\Dict\, had no success.

Thus, is there a method to set up a permanent `LaTeX User Commands.dic` maybe under another name as `myLaTeX.dic`

(2) Also the same applies for environments. Packages also provide environments.  Therefore in the spirit of (1) can one set up a permanent `myEnvironments.dic` 

regards
Anton

**Comment by Alex-Simonic** [comment]
In your Private.ini you can define extra dictionaries. For example:

```
[DICTIONARIES*]

DICTIONARY="My LaTeX Commands"
  FILE="%b\Dict\LaTeX\myLaTeX.dic"
  ENABLED=1
  MODE_FILTER="TeX"
  LOAD_ON_START=1
  SAVE_ON_EXIT=1
  ADD_NEW_WORDS=0
  USE_FOR_COMPLETION=1
  ALLOW_COMPOUNDED_WORDS=0
```
etc...

You can do the same in Dictionaries.ini (if you are not comfortable with *-sections in Private.ini). 

In AutoComplete.ini you can now add

`  DICTIONARY="My LaTeX Commands"
`

and any other custom dictionaries that you may want to create and maintain locally and use for auto-completion. No need to modify files in %B folder...

**Comment by anton-vrba** [comment]
Thanks, Works well for commands but not for environments

private.ini
DICTIONARY="My LaTeX Environments"
   FILE="%b\Dict\LaTeX\myEnvironments.dic"
   ENABLED=0
   MODE_FILTER="TeX"
   LOAD_ON_START=1
   SAVE_ON_EXIT=1
   ADD_NEW_WORDS=0
   USE_FOR_COMPLETION=1
   ALLOW_COMPOUNDED_WORDS=0
   DYNAMIC_LIST=1


in autocomplete.ini after the line
     DICTIONARY="LaTeX User Environments" // \newtheorem{foo} and 
\newenvironment{foo}
     DICTIONARY="My LaTeX Environments"

**Comment by anton-vrba** [comment]
Sorry my mistake:
ENABLED=0 should be ENABLED=1


------ Original Message ------
From "anton vrba" ***@***.***>
To "WinEdt-Team/WinEdt-Support" 
***@***.***>
Date 5/27/2024 4:28:31 PM
Subject Re: [WinEdt-Team/WinEdt-Support] Custom LaTeX.dic (Discussion 
#48)
